client-server-client/src/main/scala/zio/raft/client/ConnectionManager.scala

```diff
@@ -0,0 +1,351 @@
+package zio.raft.client
+
+import zio._
+import zio.raft.protocol._
+import scodec.bits.ByteVector
+
+/**
+ * Client connection state management and request queuing.
+ * 
+ * Manages:
+ * - Connection states (Connecting, Connected, Disconnected)
+ * - Request queuing based on connection state
+ * - Session creation and continuation
+ * - Leader redirection handling
+ * - Keep-alive mechanism
+ */
+trait ConnectionManager {
```

Comment:
should not be a trait, should be a class with one implementation. 

avoid traits when they are not needed